packages/blade/src/components/Card/CardFooter.tsx

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable @typescript-eslint/no-unnecessary-type-assertion */
+import { Divider } from './Divider';
+import { useInsideCard } from './CardContext';
+import Box from '~components/Box';
+import type { ButtonProps } from '~components/Button';
+import { Button } from '~components/Button';
+import { Text } from '~components/Typography';
+import { useBreakpoint } from '~utils';
+import { useTheme } from '~components/BladeProvider';
+
+export type CardFooterAction = Pick<
+  ButtonProps,
+  'type' | 'accessibilityLabel' | 'isLoading' | 'isDisabled' | 'icon' | 'iconPosition' | 'onClick'
+> & {
+  text: string;
+};
+
+type CardFooterProps = {
+  children?: React.ReactNode;
```

Comment:
We are not preventing anything on the top level components be it Footer, Card, CardHeader etc. We are only preventing components on the props like prefix, suffix, etc. 

IMO it's not worth the effort to be that restrictive. If they add something random going out of their way to add things on Header they might as well just use CardBody at that point.

Reply:
we do prevent in header right? using that isOfComponentId function? can't we replicate that in all the card sub-components?

Reply:
> we do prevent in header right

Yeah but only on the props. like prefix, suffix, visual etc.

> can't we replicate that in all the card sub-components?

We can, but that's what I'm saying is that it's not worth IMO. It will introduce runtime complexities too. For example in Footer we have to loop over the childrens and do the checks on every child to ensure they are the components we expect them to be.

Reply:
what i'm afraid of is that we have put the restriction in the props but this is an easy escape hatch where people will just put things they want by passing anything to Footer. Yes while can do that in Body as well and not use footer at all but if they use footer we can't do anything and people 100% will end up doing that based on the recent instances 😢